core/src/main/scala/za/co/absa/spline/harvester/dispatcher/SplineHeaders.scala

```diff
@@ -26,6 +26,7 @@ object SplineHeaders {
   // Http specific
   val AcceptRequestEncoding = s"$Prefix-Accept-Request-Encoding"
   val Timeout = "X-SPLINE-TIMEOUT"
+  val HttpSecurityKey = "X-FUNCTIONS_KEY"
```

Comment:
as I understand it X-FUNCTIONS_KEY is Azure specific header. It doesn't make sense to add one specialised header like that. Why not implement it in a away that the user can add any header they want?

Reply:
One idea to do it would be to have something like this:
```
spline.lineageDispatcher.http.header.customName.key=X-FUNCTIONS_KEY
spline.lineageDispatcher.http.header.customName.value=somekeys
```
So the user could define as many headers as they want. `customName`'s function in this is just to pair the key and the value.

Reply:
Yes , i agree with you. thanks for the idea.

Reply:
I would simplify the config a little bit by combining `key` and `value` properties in one - `key=value`.
For example:
```properties
spline.lineageDispatcher.http.header.X-FUNCTIONS_KEY=my-function-key
```

Reply:
> I would simplify the config a little bit by combining `key` and `value` properties into one `key=value` 

I was thinking about that as well, but I thought there might be cases when the header is not valid property key. But I am not aware of any such headers, so maybe I am just too paranoid here.

Reply:
RFC 9110, section [16.3.2.1](https://www.rfc-editor.org/rfc/rfc9110.html#name-considerations-for-new-field):
> While the field-name syntax is defined to allow any token character, in practice some implementations place limits on the characters they accept in field-names. **To be interoperable, new field names SHOULD constrain themselves to alphanumeric characters, "-", and ".", and SHOULD begin with a letter**. For example, the underscore ("_") character can be problematic when passed through non-HTTP gateway interfaces (see [Section 17.10](https://www.rfc-editor.org/rfc/rfc9110.html#underscore.in.fields)).

So I guess we are pretty must safe here :)